lib/bats-core/test_functions.bash

```diff
@@ -42,6 +72,37 @@ run() {
   IFS=$'\n' lines=($output)
   IFS="$origIFS"
   set "-$origFlags"
+
+  # Show results. Without quotes, multiple lines are glommed together into one
+  if [ -n "$output" ]; then
+    printf "%s\n" "$output"
+  fi
+
+  # Check exit status if requested
+  if [[ "$status" -ne 0 ]]; then
+    printf "[ rc=%d " $status
+    if [[ -n "$expected_rc" ]]; then
+      if [[ "$status" -eq "$expected_rc" ]]; then
+        printf "(expected) "
+      elif [[ $expected_rc -lt 0 ]]; then
+        printf "(expected any error)"
+        expected_rc=$status           # don't die below
+      else
+        printf "(** EXPECTED %d **) " $expected_rc;
+      fi
+    fi
+    printf "]\n"
+  fi
+
+  if [[ -n "$expected_rc" ]]; then
+    if [[ "$expected_rc" = "-1" ]]; then
+      if [[ "$status" -eq 0 ]]; then
+        die "exit code is $status; expected nonzero"
```

Comment:
I find the messages from `die` to be a stark difference from the current standard outputs. I think the current error handling already gives enough visibility to problems.